controllers/utils.go

```diff
@@ -0,0 +1,40 @@
+package controllers
+
+import (
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/client-go/discovery"
+	"sigs.k8s.io/controller-runtime/pkg/client/config"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+)
+
+// ignoreUpdateCRStatusPredicate filters update events on pulpbackup CR status
+func IgnoreUpdateCRStatusPredicate() predicate.Predicate {
+	return predicate.Funcs{
+		UpdateFunc: func(e event.UpdateEvent) bool {
+			// Ignore updates to CR status in which case metadata.Generation does not change
+			return e.ObjectOld.GetGeneration() != e.ObjectNew.GetGeneration()
+		},
+	}
+}
+
+// IsOpenShift returns true if the platform cluster is OpenShift
+func IsOpenShift() (bool, error) {
+	cfg, err := config.GetConfig()
+	if err != nil {
+		return false, err
+	}
+	client, err := discovery.NewDiscoveryClientForConfig(cfg)
+	if err != nil {
+		return false, err
+	}
+
+	_, err = client.ServerResourcesForGroupVersion("route.openshift.io/v1")
```

Comment:
Not really an issue, but Instead of using the route API
```
	_, err = client.ServerResourcesForGroupVersion("route.openshift.io/v1")
```
we could use the [APIServer](https://docs.openshift.com/container-platform/4.10/rest_api/config_apis/config-apis-index.html#apiserver-config-openshift-iov1)
```
	_, err = client.ServerResourcesForGroupVersion("config.openshift.io/v1")
```

this could be helpful in case route API gets deprecated (even though I don't believe that it will happen, at least not in the near future), which is why I think that this is not really an issue.